### PR TITLE
adding permissions check to file existence tests

### LINKED ISSUE #77: Verify file permissions as part of fileExistenceChecks
It would be great to be able to assert file permissions as part of fileExistenceChecks. Without this, it doesn't do much good to verify that a file exists if it's not readable/writable/executable/etc by the default docker user.

### PR DESCRIPTION
**Pull request by sharifelgamal**
Fixes #77 

@dlorenc @nkubala

### CODE REVIEW

File: structure_tests/file_existence_test_v1.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ type FileExistenceTestv1 struct {
 	Path        string // file to check existence of
 	IsDirectory bool   // whether or not the path points to a directory
 	ShouldExist bool   // whether or not the file should exist
+	Permissions string // expected permission string of the file
```
Comment (dlorenc) [inline structure_tests/file_existence_test_v1.go]: Could you add an example of the format of the string?

Comment (sharifelgamal) [inline structure_tests/file_existence_test_v1.go]: Done.

File: structure_tests/README.md [hunk 2]
```diff
@@ -75,7 +77,8 @@ Example:
 		"name": "Root",
```
Comment (dlorenc) [inline structure_tests/README.md]: This diff is showing up with tabs instead of spaces, could you keep it as spaces?

Comment (sharifelgamal) [inline structure_tests/README.md]: Fixed.

**Pull request merged**